shared/chat/inbox/container.js

```diff
@@ -29,14 +29,15 @@ const passesFilter = (i: Constants.InboxState, filter: I.List<string>): boolean
 const filteredInbox = createImmutableEqualSelector(
   [getInbox, getSupersededByState, getAlwaysShow, getFilter],
   (inbox, supersededByState, alwaysShow, filter) => {
-    return inbox
-      .filter(
-        i =>
-          (!i.isEmpty || alwaysShow.has(i.conversationIDKey)) &&
-          !supersededByState.get(i.conversationIDKey) &&
-          passesFilter(i, filter)
-      )
-      .map(i => i.conversationIDKey)
+    const ids = []
+    // Building a list using forEach for performance reason, only call i.conversationIDKey once
+    inbox.forEach(i => {
```

Comment:
the killer was calling .get('conversationIDKey) over and over, and yes, the time it takes really sucks (to get the same value over and over again)